beacon/validation/fields.py

```diff
@@ -292,10 +291,66 @@ async def convert(self, value: str, **kwargs) -> list:
         LOG.debug('field %s, res=%s, len(res)=%s', self.name, res, len(res))
         return res
 
-class SchemasField(ListField):
+class DatasetsField(Field):
+
+    def __init__(self, *, separator=',', trim=True, **kwargs):
+        self.separator = separator
+        self.trim = trim or True
+        super().__init__(**kwargs)
+
+    async def get_datasets(self):
+        datasets = getattr(conf, 'datasets', None)
+        if datasets is None:
+            datasets = set( [name async for _,_,name in db.fetch_datasets_access()] )
+            setattr(conf, 'datasets', datasets)
+        else:
+            LOG.debug('Using cached datasets: %s', datasets)
+        return datasets
 
     async def convert(self, value: str, **kwargs) -> (set, set):
-        values = value.split(self.separator) if value not in EMPTY_VALUES else []
-        from ..endpoints.rest.schemas import partition
-        return partition(values)
+        if value in EMPTY_VALUES:
+            return set()
 
+        values = value.split(self.separator)
+        valid, invalid = set(), set() # avoid repetitions
+        datasets = await self.get_datasets()
+        for value in values:
+            if self.trim:
+                value = value.strip()
+            if value in datasets:
+                valid.add(value)
+            else:
+                invalid.add(value)
+        invalid_size = len(invalid)
+        if invalid_size > 0:
+            error = 'are invalid datasets' if invalid_size > 1 else 'is an invalid dataset'
+            raise ValidationError(f'"{invalid}" {error}')
+        return valid
+
+class SchemaField(Field):
+
+    def __init__(self, *schemas, **kwargs):
+        if not schemas:
+            raise FieldError(self.name, 'You should select some schemas')
+        for schema in schemas:
+            if not isinstance(schema, str):
```

Comment:
do you think we could here check as well if that schema is a supported schema?

Reply:
mm I guess we could.